### ISSUE TITLE
[CodeSnippet] not showing `Show more` button if nested in `AccordionItem`

### ISSUE THREAD

**Issue by nish1013** [carbon-design-system/carbon#5316]
Example code https://codesandbox.io/s/cranky-heisenberg-8dyze

_**Steps to reproduce:**_

1. Create a `CodeSnippet` and nest inside a `AccordionItem`
2. Add a long text inside `CodeSnippet`
```
<AccordionItem>
    <CodeSnippet
        type="multi"
        showLessText="showLess"
        showMoreText="showMore"
        light
      >
        { getLongText() }
     </CodeSnippet>
</AccordionItem>
```

_**Result**_ 

Not showing the `Show more`  button

_**Expected Result**_ 

Should show `Show more` button

**Comment by emyarod** [comment]
it looks like the button is not rendering because the height is still computed as `0`, will need to look further into it

**Comment by tw15egan** [comment]
Seems like a similar problem to https://github.com/carbon-design-system/carbon/pull/5258, but this was happening in `Tabs`

**Comment by emyarod** [comment]
this seems like kind of a niche use case but for the time being I think the workaround should probably be at the application level. we need to figure out some way to recalculate the height once the accordion item content becomes visible but the solution is unclear at the moment

**Comment by asudoh** [comment]
Let's merge this to #5819; While this is addressed by #5854, I'm not sure if usage of accordion in a manner of [`<summary>`]()/[`<details>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Elemenhttps://developer.mozilla.org/en-US/docs/Web/HTML/Element/summaryt/details) tags is a correct use case of our accordion (@carbon-design-system/design please correct me if I'm wrong here)
